x-pack/plugins/uptime/public/components/overview/alerts/alert_tls.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiExpression,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiLink,
+  EuiPopover,
+  EuiSpacer,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
+import { TlsTranslations } from './translations';
+
+interface SettingsMessageExpressionPopoverProps {
+  'aria-label': string;
+  description: string;
+  id: string;
+  setAlertFlyoutVisible: (value: boolean) => void;
+  value: string;
+}
+
+const SettingsMessageExpressionPopover: React.FC<SettingsMessageExpressionPopoverProps> = ({
```

Comment:
Does it make sense to extract it into separate file?

Reply:
I've been trying to keep local dependencies in the same file until they're needed in two places. But I'm fine moving it to its own file.

Reply:
Addressed in https://github.com/elastic/kibana/pull/63913/commits/5d57988ede458fab4cacf1b19402f57024949555.